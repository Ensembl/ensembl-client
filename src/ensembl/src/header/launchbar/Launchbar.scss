@import 'src/styles/common';

.launchbar {
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid $ens-grey;
  border-top: 1px solid $ens-grey;
  padding: 8px 6px;
}

.categoriesWrapper {
  display: flex;
  flex-wrap: nowrap;
  overflow-x: auto;

  &::-webkit-scrollbar {
    display: none;
  }
}

.categories {
  display: flex;
  flex: 0 0 auto;
}

.category {
  display: flex;
  padding: 0 15px;

  &:first-child {
    padding-left: 2px;
  }

  &:not(:last-child) {
    border-right: 1px solid $ens-grey;
  }
}

.launchbarButton {
  display: inline-block;
  height: 30px;
  width: 30px;
  border: 1px solid $ens-blue;
  margin: 0 12px;

  svg {
    width: 100%;
    height: 100%;
    transform: scale(1.1);
    fill: $ens-white;
    background-color: $ens-blue;
  }

  img {
    max-width: 100%;
    max-height: 100%;
  }
}

.launchbarButtonSelected {
  border: 1px solid $ens-black;

  svg {
    fill: $ens-white;
    background-color: $ens-black;
  }
}

.launchbarButtonDisabled {
  border: 1px solid $ens-grey;

  svg {
    fill: $ens-white;
    background-color: $ens-grey;
  }
}

.about {
  display: flex;
  align-items: center;
  margin-left: 20px;

  .aboutText {
    font-size: 12px;
    margin-right: 3px;
    margin-top: 2px;
  }

  .launchbarButton {
    border: none;

    &:last-child {
      margin-right: 18px;
    }
  }
}
