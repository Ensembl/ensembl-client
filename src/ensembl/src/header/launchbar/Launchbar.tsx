import React from 'react';
import SlideDown from 'react-slidedown';

import ensemblIcon from 'static/img/launchbar/ensembl-logo.png'; // <-- note it's a png

import { ReactComponent as SearchIcon } from 'static/img/launchbar/search.svg';
import { ReactComponent as SpeciesSelectorIcon } from 'static/img/launchbar/species-selector.svg';
import { ReactComponent as BrowserIcon } from 'static/img/launchbar/browser.svg';
import { ReactComponent as VEPIcon } from 'static/img/launchbar/vep.svg';
import { ReactComponent as CustomDownloadIcon } from 'static/img/launchbar/custom-download.svg';
import { ReactComponent as HelpIcon } from 'static/img/launchbar/help.svg';
import { isEnvironment, Environment } from 'src/shared/helpers/environment';

import LaunchbarButton from './LaunchbarButton';

import styles from './Launchbar.scss';

import { CommittedItem } from 'src/content/app/species-selector/types/species-search';

type LaunchbarProps = {
  launchbarExpanded: boolean;
  committedSpecies: CommittedItem[];
};

export const getCategoryClass = (separator: boolean): string => {
  return separator ? 'border' : '';
};

const LaunchbarContent = (props: LaunchbarProps) => (
  <div className={styles.launchbar}>
    <div className={styles.categoriesWrapper}>
      <div className={styles.categories}>
        <div className={styles.category}>
          <LaunchbarButton
            app="global-search"
            description="Global search"
            icon={SearchIcon}
            enabled={false}
          />
          <LaunchbarButton
            app="species-selector"
            description="Species selector"
            icon={SpeciesSelectorIcon}
            enabled={true}
          />
        </div>
        <div className={styles.category}>
          <LaunchbarButton
            app="browser"
            description="Genome browser"
            icon={BrowserIcon}
            enabled={props.committedSpecies.length > 0}
          />
        </div>
        <div className={styles.category}>
          <LaunchbarButton
            app="tools"
            description="Tools"
            icon={VEPIcon}
            enabled={false}
          />
        </div>
        <div className={styles.category}>
          <LaunchbarButton
            app="custom-download"
            description="Downloads"
            icon={CustomDownloadIcon}
            enabled={isEnvironment([
              Environment.DEVELOPMENT,
              Environment.INTERNAL
            ])}
          />
        </div>
        <div className={styles.category}>
          <LaunchbarButton
            app="help-docs"
            description="Help & documentation"
            icon={HelpIcon}
            enabled={false}
          />
        </div>
      </div>
    </div>
    <div className={styles.about}>
      <span className={styles.aboutText}>Genome research database</span>
      <LaunchbarButton
        app="about"
        description="About Ensembl"
        icon={ensemblIcon}
        enabled={false}
      />
    </div>
  </div>
);

const Launchbar = (props: LaunchbarProps) => {
  return (
    <SlideDown transitionOnAppear={false}>
      {props.launchbarExpanded ? <LaunchbarContent {...props} /> : null}
    </SlideDown>
  );
};

export default Launchbar;
