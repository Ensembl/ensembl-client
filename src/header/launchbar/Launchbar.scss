@import 'src/styles/common';

.launchbar {
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid $grey;
  border-top: 1px solid $grey;
  padding: 0 $standard-gutter;
  height: 60px;
  align-items: center;

  @media screen and (max-width: 700px) {
    overflow-x: auto;
  }
}

.categoriesWrapper {
  display: flex;
  flex-wrap: nowrap;

  &::-webkit-scrollbar {
    display: none;
  }
}

.categories {
  display: flex;
  flex: 0 0 auto;
}

.launchbarButton {
  display: inline-block;
  height: 32px;
  width: 32px;

  button {
    width: 32px;
    height: 32px;
  }
}

.category {
  display: flex;
  column-gap: 20px;
  padding: 0 $standard-gutter;

  &:first-child {
    padding-left: 0px;

    .launchbarButton {
      &:first-child {
        margin-left: 0;
      }
    }
  }

  &:not(:last-child) {
    border-right: 1px solid $grey;
  }
}

.launchbarButtonImage {
  background-color: $blue;

  svg {
    fill: $white;
  }
}

.launchbarButtonSelectedImage {
  background-color: $black;

  svg {
    fill: $white;
  }
}

.launchbarButtonDisabledImage {
  background-color: $grey;

  svg {
    fill: $white;
  }

  button {
    cursor: default;
  }
}

.aboutEnsembl {
  display: flex;
  align-items: center;
  color: $blue;
}

.logotype {
  height: 10px;
  fill: $blue;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

.toolsIconWrapper {
  position: relative;
}

.notification {
  border-radius: 10px;
  height: var(--launchbar-notification-size, 18px);
  width: var(--launchbar-notification-size, 18px);
  position: absolute;
  top: 0;
  right: 0;
  transform: translate(50%, -50%);

  &Red {
    background-color: $red;
  }

  &Green {
    background-color: $green;
  }
}
