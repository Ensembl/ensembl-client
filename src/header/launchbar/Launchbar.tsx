/**
 * See the NOTICE file distributed with this work for additional information
 * regarding copyright ownership.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Link } from 'react-router-dom';

import useHeaderAnalytics from '../hooks/useHeaderAnalytics';

import {
  GenomeBrowserIcon,
  GlobalSearchIcon,
  HelpIcon
} from 'src/shared/components/app-icon';
import LaunchbarButton from './LaunchbarButton';
import SpeciesSelectorLaunchbarButton from './SpeciesSelectorLaunchbarButton';
import EntityViewerLaunchbarButton from './EntityViewerLaunchbarButton';
import BlastLaunchbarButton from './BlastLaunchbarButton';

import Logotype from 'static/img/brand/logotype.svg';

import styles from './Launchbar.module.css';

export const getCategoryClass = (separator: boolean): string => {
  return separator ? 'border' : '';
};

const Launchbar = () => {
  return (
    <div className={styles.launchbar}>
      <div className={styles.categoriesWrapper}>
        <div className={styles.categories}>
          <div className={styles.category}>
            <LaunchbarButton
              path="/global-search"
              description="Site search"
              icon={GlobalSearchIcon}
              enabled={false}
            />
            <SpeciesSelectorLaunchbarButton />
          </div>
          <div className={styles.category}>
            <LaunchbarButton
              path="/genome-browser"
              description="Genome browser"
              icon={GenomeBrowserIcon}
              enabled={true}
            />
          </div>
          <div className={styles.category}>
            <EntityViewerLaunchbarButton />
          </div>
          <div className={styles.category}>
            <BlastLaunchbarButton />
          </div>
          <div className={styles.category}>
            <LaunchbarButton
              path="/help"
              description="Help & documentation"
              icon={HelpIcon}
              enabled={true}
            />
          </div>
        </div>
      </div>
      <AboutEnsembl />
    </div>
  );
};

const AboutEnsembl = () => {
  const { trackLaunchbarAppChange } = useHeaderAnalytics();

  return (
    <Link
      to="/about"
      className={styles.aboutEnsembl}
      onClick={() => trackLaunchbarAppChange('About')}
    >
      About the
      <Logotype className={styles.logotype} />
      project
    </Link>
  );
};

export default Launchbar;
