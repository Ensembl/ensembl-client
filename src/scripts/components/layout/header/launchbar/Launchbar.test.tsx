import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  MemoryRouter,
  RouteComponentProps,
  withRouter
} from 'react-router-dom';

import { Launchbar } from './Launchbar';

describe('<Launchbar />', () => {
  let wrapper: any;

  beforeEach(() => {
    const changeCurrentApp: (currentApp: string) => void = jest.fn();

    const WrappedComponent: any = withRouter((props: RouteComponentProps) => (
      <Launchbar
        changeCurrentApp={changeCurrentApp}
        currentApp={''}
        launchbarExpanded={true}
        {...props}
      />
    ));

    wrapper = render(
      <MemoryRouter>
        <WrappedComponent />
      </MemoryRouter>
    );
  });

  test('renders about icon separate of other categories', () => {
    expect(wrapper.find('.launchbar > .categories-wrapper')).toHaveLength(1);
    expect(wrapper.find('.launchbar > .about')).toHaveLength(1);
  });

  test('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
