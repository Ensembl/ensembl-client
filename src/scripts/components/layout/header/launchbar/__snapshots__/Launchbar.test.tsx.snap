// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Launchbar /> renders correctly 1`] = `
<ForwardRef
  transitionOnAppear={false}
>
  <div
    className="launchbar"
  >
    <div
      className="categories-wrapper"
    >
      <div
        className="categories"
      >
        <dl
          className="border"
          key="search"
        >
          <dt
            className="global-search-app-icon"
            key="global-search"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "global search",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "global-search",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="species-selector-app-icon"
            key="species-selector"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "species selector",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "species-selector",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className="border"
          key="browsers"
        >
          <dt
            className="browser-app-icon"
            key="browser"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "browser",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "browser",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className="border"
          key="tools"
        >
          <dt
            className="blast-app-icon"
            key="blast"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "BLAST/BLAT",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "blast",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="vep-app-icon"
            key="vep"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "VEP",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "vep",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="conversions-app-icon"
            key="conversions"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "Conversions",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "conversions",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="ld-app-icon"
            key="ld"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "LD calculator",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "ld",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="biomart-app-icon"
            key="biomart"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "biomart",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "biomart",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className="border"
          key="downloads"
        >
          <dt
            className="bulk-download-app-icon"
            key="bulk-download"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "bulk download",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "bulk-download",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="custom-download-app-icon"
            key="custom-download"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "custom download",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "custom-download",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className=""
          key="learning"
        >
          <dt
            className="help-docs-app-icon"
            key="help-docs"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "help and documentation",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "help-docs",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="glossary-app-icon"
            key="glossary"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "glossary",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "glossary",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="new-info-app-icon"
            key="new-info"
          >
            <LaunchbarIcon
              app={
                Object {
                  "description": "what is new and what has changed",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "new-info",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
      </div>
    </div>
    <div
      className="about"
    >
      <dl>
        <dt
          className="about-app-icon"
        >
          <LaunchbarIcon
            app={
              Object {
                "description": "about ensembl",
                "icon": Object {
                  "default": Object {},
                  "selected": Object {},
                },
                "name": "about",
              }
            }
            currentApp=""
          />
        </dt>
      </dl>
    </div>
  </div>
</ForwardRef>
`;
